compiler/vm.nim

```diff
@@ -547,9 +551,12 @@ proc rawExecute(c: PCtx, start: int, tos: PStackFrame): TFullReg =
         "pc", $pc, "opcode", alignLeft($c.code[pc].opcode, 15),
         "ra", regDescr("ra", ra), "rb", regDescr("rb", instr.regB),
         "rc", regDescr("rc", instr.regC)]
-
+    if hasKeyCached(c.config, "nimVmTrace", symbolsCacheId, nimVmTrace):
```

Comment:
If a call with no less than 4 args before every executed opcode has no performance impact then all that means is that our VM is already too slow... And it gets harder and harder to ever optimize code that had this laissez-faire attitude to performance.

Reply:
``if c.config.vmTrace :``

Reply:
> If a call with no less than 4 args before every executed opcode has no performance impact 

actually I meant to turn `hasKeyCached` into a template, in which case the performance argument still holds, because it reduces to this: `if symbolsCacheId != config.symbolsCacheId:` which branch predictor can predict as almost always false. The only cases where this is violated is where `symbols` is updated, a rare event.

But this will disappear anyway if I use an API call instead of {.define.}

Reply:
now using a vmops